### PR TITLE
Feature/issue 201 navigate between channels

### PR DESCRIPTION
**Pull request by tungnk123**

### CODE REVIEW

File: feature/channel/src/main/java/com/m3u/feature/channel/ChannelViewModel.kt [hunk 1]
```diff
@@ -81,6 +85,23 @@ class ChannelViewModel @Inject constructor(
     internal val channel: StateFlow<Channel?> = playerManager.channel
     internal val playlist: StateFlow<Playlist?> = playerManager.playlist
 
+    val adjacentChannels: StateFlow<AdjacentChannels?> = flatmapCombined(
+        playlist.filterNotNull(),
+        channel.filterNotNull()
```
Comment (oxyroid) [inline feature/channel/src/main/java/com/m3u/feature/channel/ChannelViewModel.kt:90]: Don't use `filterNotNull` here. If these upstream flows emit null it means they are cleared. So the adjacentChannels should be cleared as well.

File: feature/channel/src/main/java/com/m3u/feature/channel/ChannelMask.kt [hunk 2]
```diff
@@ -542,11 +561,13 @@ private fun MaskCenterButton(
...
-    Box(modifier, contentAlignment = Alignment.Center) {
+    Box(
+        modifier,
+        contentAlignment = Alignment.Center
+    ) {
         when (maskCenterState) {
             Replay, Play, Pause -> {
-                MaskCircleButton(
-                    state = maskState,
+                MaskCircleButton(state = maskState,
```
Comment (oxyroid) [inline feature/channel/src/main/java/com/m3u/feature/channel/ChannelMask.kt:570]: line break.

Comment (tungnk123) [reply feature/channel/src/main/java/com/m3u/feature/channel/ChannelMask.kt:570]: image
Why i cannot use the reformat feature with these file, when i use it don't move the `state` to the next line? 
I attached the image from my Code Style Settings

File: feature/channel/src/main/java/com/m3u/feature/channel/ChannelViewModel.kt (hunk 1 above)
Comment (tungnk123) [reply feature/channel/src/main/java/com/m3u/feature/channel/ChannelViewModel.kt:90]: The issue occurs in the usage of flatmapCombined, where the expected input type is Any, but a value of type Playlist? is being passed.
image

Comment (oxyroid) [reply feature/channel/src/main/java/com/m3u/feature/channel/ChannelViewModel.kt:90]: Maybe we should adjust the method again.

File: feature/channel/src/main/java/com/m3u/feature/channel/ChannelMask.kt (hunk 2 above)
Comment (oxyroid) [reply feature/channel/src/main/java/com/m3u/feature/channel/ChannelMask.kt:570]: the idea reformation would make mistakes sometimes.

**Pull request closed without merging**